### PR TITLE
Accept arbitrary incoming attachments.

### PR DESCRIPTION
**Pull request by charlesmchen**
PTAL @michaelkirk

### CODE REVIEW

File: src/Messages/Attachments/OWSAttachmentsProcessor.m [hunk 1]
```diff
@@ -79,7 +79,10 @@ - (instancetype)initWithAttachmentProtos:(NSArray<OWSSignalServiceProtosAttachme
 
         [attachmentIds addObject:pointer.uniqueId];
 
-        if ([MIMETypeUtil isSupportedMIMEType:pointer.contentType]) {
+        // TODO: We need to revisit isSupportedMIMEType.
+        if ([MIMETypeUtil isSupportedMIMEType:pointer.contentType] ||
+            [pointer.contentType isEqualToString:OWSMimeTypeOversizeTextMessage] ||
+            [pointer.contentType isEqualToString:OWSMimeTypeUnknownForTests]) {
```
Comment (charlesmchen) [inline src/Messages/Attachments/OWSAttachmentsProcessor.m:85]: When we want to allow users to send arbitrary outgoing attachments as well, we should rework `isSupportedMIMEType`.

File: src/Util/MIMETypeUtil.h [hunk 2]
```diff
@@ -52,4 +53,8 @@ extern NSString *const OWSMimeTypeOversizeTextMessage;
 + (NSSet<NSString *> *)supportedImageUTITypes;
 + (NSSet<NSString *> *)supportedAnimatedImageUTITypes;
 
++ (NSString *)utiTypeForMIMEType:(NSString *)mimeType;
++ (NSString *)fileExtensionForUTIType:(NSString *)utiType;
++ (NSString *)fileExtensionForMIMEType:(NSString *)mimeType;
```
Comment (charlesmchen) [inline src/Util/MIMETypeUtil.h:58]: These functions are used in the new "arbitrary attachments" changes to Signal-iOS.

File: src/Util/MIMETypeUtil.m [hunk 3]
```diff
@@ -13,6 +13,7 @@
 NSString *const OWSMimeTypeApplicationOctetStream = @"application/octet-stream";
 NSString *const OWSMimeTypeImagePng = @"image/png";
 NSString *const OWSMimeTypeOversizeTextMessage = @"text/x-signal-plain";
+NSString *const OWSMimeTypeUnknownForTests = @"unknown/mimetype";
```
Comment (michaelkirk) [inline src/Util/MIMETypeUtil.m:16]: How are you using this for testing?

**Review by michaelkirk — APPROVED** [review]
LGTM

File: src/Util/MIMETypeUtil.m (hunk 3 above)
Comment (charlesmchen) [reply src/Util/MIMETypeUtil.m:16]: In the Debug UI, there will soon be an action to send an attachment with this MIME type, which should always be "unknown" to the recipient.

**Pull request merged**